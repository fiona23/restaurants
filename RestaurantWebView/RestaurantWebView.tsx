import React from 'react';
import {StyleSheet, View, Button} from 'react-native';
import {WebView} from 'react-native-webview';

interface Props {
  displayWebView: string;
  setDisplayWebView: (url: string) => void;
}
export const RestaurantWebView = ({
  displayWebView,
  setDisplayWebView,
}: Props) => {
  return (
    <>
      <View style={styles.backButton}>
        <Button
          title="Back"
          onPress={() => setDisplayWebView('')}
          accessibilityLabel="Press to go back"
        />
      </View>
      <WebView source={{uri: displayWebView}} />
    </>
  );
};

const styles = StyleSheet.create({
  backButton: {
    marginTop: 40,
    marginLeft: 10,
    flexDirection: 'row',
  },
});
