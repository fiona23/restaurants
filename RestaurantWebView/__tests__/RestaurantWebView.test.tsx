import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {RestaurantWebView} from '../RestaurantWebView';

describe('RestaurantWebView', () => {
  it('should renders correctly', () => {
    const {toJSON} = renderer.create(
      <RestaurantWebView
        displayWebView={'url'}
        setDisplayWebView={() => jest.fn()}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
