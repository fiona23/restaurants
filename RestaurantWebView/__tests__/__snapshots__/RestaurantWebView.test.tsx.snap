// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantWebView should renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 10,
        "marginTop": 40,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Back
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "uri": "url",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
        ]
      }
      useSharedProcessPool={true}
    />
  </View>,
]
`;
